lib/topological_inventory/orchestrator/metric_scaler/prometheus_watcher.rb

```diff
@@ -0,0 +1,105 @@
+require "topological_inventory/orchestrator/metric_scaler/watcher"
+
+module TopologicalInventory
+  module Orchestrator
+    class MetricScaler
+      class PrometheusWatcher < Watcher
+        METRICS_CHECK_INTERVAL = 150
+        METRICS_RANGE          = "10m"
+        METRICS_STEP           = "1m"
+
+        def initialize(deployment_config, deployment_config_name, prometheus_host, logger)
+          super(deployment_config, deployment_config_name, logger)
+          @prometheus_host = prometheus_host
+          @scaling_allowed.value = false
+        end
+
+        def configured?
+          @query && @max_replicas && @min_replicas && @target_usage && @scale_threshold
+        end
+
+        def configure
+          @max_replicas        = deployment_config.metadata.annotations["metric_scaler_max_replicas"]&.to_i        # i.e. "5"
+          @min_replicas        = deployment_config.metadata.annotations["metric_scaler_min_replicas"]&.to_i        # i.e. "1"
+          @target_usage        = deployment_config.metadata.annotations["metric_scaler_target_usage_pct"]&.to_f    # i.e. "50"
+          @scale_threshold     = deployment_config.metadata.annotations["metric_scaler_scale_threshold_pct"]&.to_f # i.e. "20"
```

Comment:
`@target_usage` and `@scale_threshold` won't be defined as a percentage value but a size of the Kafka lag, so I would omit the `_pct` suffix